hs_core/models.py

```diff
@@ -113,7 +120,7 @@ def can_change(self, request):
     def can_view(self, request):
         user = get_user(request)
 
-        if self.raccess.public or self.raccess.discoverable:
+        if self.raccess.public or self.raccess.discoverable or self.raccess.immutable:
```

Comment:
Removed immutable flag from this conditional checking. Agreed it is not needed here.